pennylane_lightning/core/src/simulators/lightning_kokkos/gates/tests/Test_StateVectorKokkos_NonParam.cpp

```diff
@@ -758,43 +801,281 @@ TEMPLATE_TEST_CASE("StateVectorKokkos::applyMultiQubitOp",
+
+            for (std::size_t j = 0; j < exp2(num_qubits); j++) {
+                CHECK(imag(sv_gate_host[j]) ==
+                      Approx(imag(sv_control_host[j])));
+                CHECK(real(sv_gate_host[j]) ==
+                      Approx(real(sv_control_host[j])));
             }
         }
     }
+
+    SECTION("N-controlled PauliZ ") {
+        if (control != wire) {
+            Kokkos::deep_copy(sv_gate.getView(), ini_sv);
+            Kokkos::deep_copy(sv_control.getView(), ini_sv);
+
+            sv_gate.applyOperation("CZ", {control, wire}, inverse);
+            sv_control.applyOperation(
+                "PauliZ", std::vector<std::size_t>{control},
+                std::vector<bool>{true}, std::vector<std::size_t>{wire});
+            auto sv_gate_host = Kokkos::create_mirror_view_and_copy(
+                Kokkos::HostSpace{}, sv_gate.getView());
+            auto sv_control_host = Kokkos::create_mirror_view_and_copy(
+                Kokkos::HostSpace{}, sv_control.getView());
+
+            for (std::size_t j = 0; j < exp2(num_qubits); j++) {
+                CHECK(imag(sv_gate_host[j]) ==
+                      Approx(imag(sv_control_host[j])));
+                CHECK(real(sv_gate_host[j]) ==
+                      Approx(real(sv_control_host[j])));
+            }
+        }
+    }
+}
+
+TEMPLATE_TEST_CASE("StateVectorKokkos::applyOperation non-param "
+                   "one-qubit with multiple-controls",
+                   "[StateVectorKokkos_NonParam]", float, double) {
+    using ComplexT = StateVectorKokkos<TestType>::ComplexT;
+    const bool inverse = false;
+    const std::size_t num_qubits = 3;
+    StateVectorKokkos<TestType> kokkos_sv{num_qubits};
+
+    kokkos_sv.applyOperations({{"Hadamard"}, {"Hadamard"}, {"Hadamard"}},
+                              {{0}, {1}, {2}}, {{false}, {false}, {false}});
+
+    auto ini_sv = Kokkos::create_mirror_view_and_copy(Kokkos::HostSpace{},
+                                                      kokkos_sv.getView());
+    StateVectorKokkos<TestType> sv_gate{num_qubits};
+    StateVectorKokkos<TestType> expected_result{num_qubits};
+
+    SECTION("2-controlled PauliY") {
+        Kokkos::deep_copy(sv_gate.getView(), ini_sv);
+
+        const std::vector<std::size_t> control_wires = {0, 2};
+        const std::vector<bool> control_values = {true, false};
+        const std::vector<std::size_t> target_wire = {1};
+        sv_gate.applyOperation("PauliY", control_wires, control_values,
+                               target_wire, inverse);
+        auto sv_gate_host = Kokkos::create_mirror_view_and_copy(
+            Kokkos::HostSpace{}, sv_gate.getView());
+
+        std::vector<ComplexT> expected_result{
+            ComplexT{0.35355339, 0.0},  ComplexT{0.35355339, 0.0},
+            ComplexT{0.35355339, 0.0},  ComplexT{0.35355339, 0.0},
+            ComplexT{0.0, -0.35355339}, ComplexT{0.35355339, 0.0},
+            ComplexT{0.0, 0.35355339},  ComplexT{0.35355339, 0.0},
```

Comment:
Could you explain where come this number and if it is possible to generate it analytically, please?

The same for the rest of the test

Reply:
These are generated from Pennylane. For this case it is possible to generate anaylytically, but for some gates (those with rotations) these are just numerical. I will add a comment in the code.